manic/sourcetree.py

```diff
@@ -331,11 +339,23 @@ def checkout(self, verbosity, load_all, load_comp=None):
             printlog('Checking out externals: ', end='')
 
         if load_all:
-            load_comps = self._all_components.keys()
+            tmp_comps = self._all_components.keys()
         elif load_comp is not None:
-            load_comps = [load_comp]
+            tmp_comps = [load_comp]
         else:
-            load_comps = self._required_compnames
+            tmp_comps = self._required_compnames
+
+        load_comps = []
+        for comp in tmp_comps:
+            prereq = self._all_components[comp].get_prereq()
+            if prereq:
+                if prereq in self._all_components and prereq not in load_comps:
+                    load_comps.append(prereq)
```

Comment:
Unless I'm missing something, this doesn't check if the given prereq itself has prerequisites. I'm thinking this should be checked recursively. In doing so, there should probably be an error raised if a circularity of prerequisites is found.